File: src/main/java/org/jukeboxmc/network/handler/EntityEventHandler.java
package org.jukeboxmc.network.handler;

import com.nukkitx.protocol.bedrock.data.entity.EntityEventType;
import com.nukkitx.protocol.bedrock.packet.EntityEventPacket;
import org.jetbrains.annotations.NotNull;
import org.jukeboxmc.Server;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class EntityEventHandler implements PacketHandler<EntityEventPacket>{

    @Override
    public void handle(@NotNull EntityEventPacket packet, @NotNull Server server, @NotNull Player player ) {
        if ( packet.getType().equals( EntityEventType.EATING_ITEM ) ) {
            if ( packet.getData() == 0 || packet.getRuntimeEntityId() != player.getEntityId() ) {
                return;
            }
            server.broadcastPacket( packet );
        }
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/network/handler/EntityEventHandler.kt
package org.jukeboxmc.server.network.handler

import org.cloudburstmc.protocol.bedrock.data.entity.EntityEventType
import org.cloudburstmc.protocol.bedrock.packet.EntityEventPacket
import org.jukeboxmc.server.JukeboxServer
import org.jukeboxmc.server.player.JukeboxPlayer

class EntityEventHandler : PacketHandler<EntityEventPacket> {

    override fun handle(packet: EntityEventPacket, server: JukeboxServer, player: JukeboxPlayer) {
        if (packet.type == EntityEventType.EATING_ITEM) {
            if (packet.data == 0 || packet.runtimeEntityId != player.getEntityId()) {
                return
            }
            server.broadcastPacket(packet)
        }
    }
}